### PR TITLE
Optimize recover wallets funds script

### PR DESCRIPTION
**Pull request by haythemsellami**
The script used to send back ETH from users wallets back to Coinsence wallets require a lot of memory to run.
In this PR, trying to implement [similar mechanism](https://docs.ethers.io/ethers.js/html/cookbook-accounts.html#sweep-an-account-into-another) and to run the transfer instructions sequentially.

### CODE REVIEW

File: apps/coinsence-kit/contracts/CoinsenceKit.sol [hunk 1]
```diff
@@ -73,8 +73,8 @@ contract CoinsenceKit is KitBase {
         Coin coin;
 
         bytes32[2] memory appIds = [
-            apmNamehashOpen("coinsence-space"),     // 0
-            apmNamehashOpen("coinsence-coin")       // 1
+            apmNamehash("coinsence-space"),     // 0
+            apmNamehash("coinsence-coin")       // 1
```
Comment (bumi) [inline apps/coinsence-kit/contracts/CoinsenceKit.sol]: attention, attention! :) 
we should really fix this as the next thing...

File: lib/http/utils/load-wallet.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ const path = require('path');
 
 const keysDirectory = process.env.COINSENCE_WALLETS_DIR || path.join(__dirname, '../../../', 'keys');
 
-module.exports = function (accountId, password) {
+module.exports = async function (accountId, password) {
```
Comment (bumi) [inline lib/http/utils/load-wallet.js]: why is this needed here?

**Review by bumi — CHANGES_REQUESTED** [review]
ok, this looks nice! and if I read it correctly it does not run the transactions in parallel but one by one?

File: apps/coinsence-kit/contracts/CoinsenceKit.sol (hunk 1 above)
Comment (haythemsellami) [reply apps/coinsence-kit/contracts/CoinsenceKit.sol]: yes we should :\ I forget to change it every time I deploy the kit locally XD

File: lib/http/utils/load-wallet.js (hunk 2 above)
Comment (haythemsellami) [reply lib/http/utils/load-wallet.js]: mmm to use `await` here https://github.com/Coinsence/coinsence-monorepo/pull/29/files#diff-acff7027fa464ebffb6774cfb33669cbR40 to wait for the returned promise...
what do u think ?

Comment (bumi) [reply lib/http/utils/load-wallet.js]: I think that is not needed, this function returns a promise anyway. and you need an async on the function that is using await.

**Comment by haythemsellami** [comment]
> ok, this looks nice! and if I read it correctly it does not run the transactions in parallel but one by one?

Yes

File: scripts/recover-funds.js [hunk 3]
```diff
@@ -2,42 +2,43 @@ const fs = require('fs');
...
-  }
-});
-    
+    }
+  });
+  
+});    
+
+async function transferFund(accountId, password, provider, newAddress) {
+  let wallet = await loadWallet(accountId, password);
```
Comment (bumi) [inline scripts/recover-funds.js]: we should do some exception handling here, as this might raise an error if the wallet file can not be loaded.

File: scripts/recover-funds.js [hunk 4]
```diff
@@ -2,42 +2,43 @@ const fs = require('fs');
...
-        console.log(e);
-        new Error(`Can't access wallet file: ${accountId}`);
+  fs.readdir(keysDirectory, function (err, accounts) {
+    if (err) {
+      return console.log('Error fetching wallets: ' + err);
+    }
+    else {
+      accounts.forEach(function (account) {
+        let accountId = path.basename(account, '.json');
+        transferFund(accountId, password, ethProvider, coinsenceWallet);
```
Comment (bumi) [inline scripts/recover-funds.js]: I think we do need an await here?

**Review by bumi — CHANGES_REQUESTED** [review]

**Review by bumi — APPROVED** [review]
